addons/hw_drivers/controllers/driver.py

```diff
@@ -560,19 +560,27 @@ def run(self):
 
 class SocketManager(Thread):
 
+    def __init__(self):
+        super(SocketManager, self).__init__()
+        self.open_socket(9000)
+
+    def open_socket(self, port):
+        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        self.sock.bind(('', port))
+        self.sock.listen(1)
+        self.sock.setblocking(False)
```

Comment:
remove setblocking and set defaulttimeout

Reply:
set it to 15s

Reply:
A consequence of setting the timeout will be that the sockets created by `accept` will also have that timeout: https://docs.python.org/3/library/socket.html#timeouts-and-the-accept-method . Is that intentional? At present the sockets created by `accept` are always blocking with no timeout (checked the OS-dependent part on my machine and on a box).

Reply:
@raf-odoo it was the same with setblocking(false) as it is equal to settimeout(0.0). What would be the problem of the timout in a while true loop?

Reply:
With `setblocking(False)` on `self.sock` (as in the current PR), the sockets created by `self.sock.accept()` would still be blocking and won't timeout in my current understanding. In other words, if you do a `recv()` on one of the created sockets, it will just block on that until data arrives.

If a timeout is set to 15 seconds on the listening socket (`self.sock` in this case), the new sockets created from the `accept` will also have that timeout. Meaning the `recv()` I mentioned will raise a timeout exception if it gets no data within that time. If the code of the drivers is not expecting that, this could result in issues. That's why I wanted to check with you if that impact is expected.

The reason I started asking about the non-blocking accept and the sleep in `SocketManager.run` is because I don't understand the reason for it. Why are these changes being made? The Ingenico related code has been plagued by instability issues for months. In its current form this PR together with https://github.com/odoo/enterprise/pull/16099 was intended to be a fix for that, but it has made the reported instabilities more frequent so far. In order to gain confidence that we're moving in the right direction, it would help if there was some context on why some of the changes are being made and what the expected effect is. That will also help with being able to reproduce the issues the customers are experiencing. There appear to be timing / randomness factors involved, like last year. Hence my interest when changes like this are made, because they can either mask existing problems or create new ones if we're not careful.

Reply:
I see the updated code uses `settimeout` instead of `setdefaulttimeout`. Was that intentional?

Reply:
@raf-odoo I suggest you to go to the office check with a terminal in a real use case, instead of "Guessing" how it will work.

Reply:
@pimodoo Am I not supposed to review this PR?

Reply:
@pimodoo 
I don't think Raf is guessing things, but more suggesting possible solutions/explanations based on what he thoroughly analyzed from the customers' data. 

Can we stay constructive and move on to this fix?
We have some customers paying and waiting for a solution so it would be best to avoid losing more time and minimize back-and-forths with experimental fixes (even if we sometimes do not have really any alternative)
It is not that easy to just "reproduce" this kind of issue, and it often requires a particular setup or specific concurrent conditions that cannot easily be mimicked.

You are the expert and we trust you, and if you are not confident about Raf's suggestions it's OK for me but we should try to merge the fix if it looks good to you this way and see the results.

Reply:
To be clear: there's currently no indication the timeout stuff is related to the customer issues. It's just that it seemed odd to me, so I pointed it out and tried to understand why.

Reply:
The problem is very complex insofar as 3 people of our team tested the PRs and they did not present the problems encountered by the customers.
So far, none of our tests reproduce the customer's error.
And the customer no longer wants to test ...

It is therefore necessary to reproduce the error with a specific flow and adapt this PR before merging it.

Reply:
I think in the case of the issue at https://github.com/odoo/odoo/pull/65456#discussion_r598553232 they might be more willing to test the fix, since there we have an explanation for what can go wrong on our end. Ideally we'd indeed also have a test case to reproduce the Ingenico behavior on our end. I'll try to go over the specs and documentation to see if there's something that could explain it. The actual network traffic on the box side we already have and in theory that could be adapted and replayed to reproduce the issue, but it wouldn't explain yet why it's happening on the Ingenico side.